### DISCUSSION TITLE
duplicate mount cache

### DISCUSSION THREAD

**Discussion by zemul** [seaweedfs/seaweedfs#3811] [General]
https://github.com/seaweedfs/seaweedfs/blob/b7de4a967e14f4d7419378e1daf590d3ef8c3c82/weed/mount/weedfs.go#L193

In the k8s environment, option.MountDirectory Is dynamic,  duplicate cache exists at the same filerPath
Is there any problem with removing the option.MountDirectory?

**Comment by zemul** [comment]
Is it safe to share cache if multiple  mounts mount the same filerDirectory? @chrislusf

**Reply by chrislusf** [reply]
Not safe with multiple cache writers.

**Reply by zemul** [reply]
> Not safe with multiple cache writers.

I understand,thank you for your answer.

**Reply by chrislusf** [reply]
if there are multiple mount on the same filer, this would cause problem though.

`option.MountDirectory` needs to be add back.

**Comment by zemul** [comment]
I seem to have found a better solution.
https://github.com/seaweedfs/seaweedfs-csi-driver/pull/68
